api/v1/controllers/samples.js

```diff
@@ -128,6 +144,7 @@ module.exports = {
    */
   putSample(req, res, next) {
     utils.noReadOnlyFieldsInReq(req, helper);
+    rejectIfNameInBody(req.body);
     doPut(req, res, next, helper);
```

Comment:
you should be able to pass in the fields that you want to check to the "noReadIOnlyFieldsInReq" function. So, for upsert and bulkUpsert do not pass the name filed to the function.